submissions/loisAdegbohungbe/DOM/signup.js

```diff
@@ -0,0 +1,92 @@
+let  form = document.getElementById('signup-form');
+let  firstName = document.getElementById('firstName');
+let  lastName = document.getElementById('lastName');
+let  phoneNumber = document.getElementById('phone');
+let  email = document.getElementById('email');
+let  password = document.getElementById('password');
+let  confirmPassword = document.getElementById('confirmPassword');
+
+validation = (field, regExp, error1, error2, event) => {
+  event.preventDefault();
+  if (field.value === '') {
+    field.nextElementSibling.textContent = error1;
+    event.preventDefault();
+    return false;
+  } else if (!field.value.match(regExp)) {
+    field.nextElementSibling.textContent = error2;
+    event.preventDefault();
+    return false;
+  } else {
+    field.nextElementSibling.textContent = '';
+    return true;
+  }
+}
+
+confirmPasswordCheck = (passwordField, confirmPasswordField, error1, error2,  event) => {
+    event.preventDefault();
+    if (confirmPasswordField.value === '') {
+        confirmPasswordField.nextElementSibling.textContent = error1;
+        event.preventDefault();
+        return false;
+    }
+    else if (passwordField.value !=confirmPasswordField.value) {
+      confirmPasswordField.nextElementSibling.textContent = error2;
+      event.preventDefault();
+      return false;
+    } else {
+      confirmPasswordField.nextElementSibling.textContent = '';
+      return true;
+    }
+  }
+
+formValidation = (event) => {
+  const nameRegex = /^[A-Za-z0-9 ]+$/,
+    phoneRegex = /^[\+]?[0-9]{8,14}$/i,
+    emailRegex = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,10})+$/,
+    passwordRegex = /(?=.*\d)(?=.*[a-z])(?=.*[a-zA-Z]).{6,}$/;
+  validation(
+    firstName,
+    nameRegex,
+    'Please enter your First Name',
+    'Please enter a valid First Name ',
+    event
```

Comment:
@damlois Why are you passing in event as the last parameter?

Reply:
You only need it in the function

Reply:
It won't work if not passed @ToluAkin